base/range.jl

```diff
@@ -144,7 +143,7 @@ function steprange_last(start::T, step, stop) where T
             last = steprange_last_empty(start, step, stop)
         else
             diff = stop - start
-            if T<:Signed && (diff > zero(diff)) != (stop > start)
```

Comment:
Yes, I agree, this seems a little strange.  I understand why you need to remove this — and I agree it's a good thing to do, but I think we want to preserve the effective behavior here.

Perhaps we should detected (un)signed-ness based the behavior of `stop - start > zero(diff) && start - stop > zero(diff)`?